zigzag/classes/hardware/architecture/DimcArrayUnit.py

```diff
@@ -0,0 +1,678 @@
+import numpy as np
+import math
+import copy
+if __name__ == "__main__":
+    from imc_unit import ImcUnit
+    import logging as _logging
+    _logging_level = _logging.INFO
+    _logging_format = '%(asctime)s - %(funcName)s +%(lineno)s - %(levelname)s - %(message)s'
+    _logging.basicConfig(level=_logging_level,
+                         format=_logging_format)
+else:
+    import logging as _logging
+    from zigzag.classes.hardware.architecture.imc_unit import ImcUnit
+
+###############################################################################################################
+# README
+#   . class DimcArrayUnit (defines the energy/area/delay cost of a DIMC array)
+# How to use this file?
+#   . This file is internally called in ZigZag-IMC framework.
+#   . It can also be run independently, for mainly debugging. An example is given at the end of the file.
+###############################################################################################################
+
+class DimcArrayUnit(ImcUnit):
```

Comment:
Same comment as for AimcArrayUnit: Why is it called "ArrayUnit"?

Reply:
Renamed to D/AimcArray.